pgsql-flexible-server.tf

```diff
@@ -44,6 +59,11 @@ resource "azurerm_postgresql_flexible_server" "pgsql_server" {
 
   sku_name = var.pgsql_sku
 
+  authentication {
```

Comment:
(this got hidden when the line I commented on was changed before)

could adding a user role be added to the database as well, in cnp-module-postgres we do it with:

https://github.com/hmcts/cnp-module-postgres/blob/postgresql_tf/set-postgres-permissions.bash

but I think it should be doable with the postgres provider as well (could be a follow up pull request / story)

example provider usage:
https://github.com/hmcts/pip-data-management/blob/e3fed2f72fa8731679537fd3a8d8319a4394ebee/infrastructure/tf-db-main.tf#L27-L42